engines/cryomni3d/sprites.cpp

```diff
@@ -0,0 +1,230 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "common/debug.h"
+#include "common/file.h"
+#include "graphics/surface.h"
+
+#include "engines/cryomni3d/sprites.h"
+
+// #define SPRTIES_DEBUG
+
+namespace CryOmni3D {
+
+#define MAP_ID(id) \
+    do { \
+        if (_map) { \
+            id = (*_map)[id]; \
+        } \
+    } while(false)
+
+Sprites::Sprites() : _map(nullptr) {
+	_surface = new Graphics::Surface();
+}
+
+Sprites::~Sprites() {
+	for (Common::Array<CryoCursor *>::iterator it = _cursors.begin(); it != _cursors.end(); it++) {
+		if ((*it)->refCnt > 1) {
+			(*it)->refCnt--;
+		} else {
+			delete *it;
+		}
+	}
+	delete _map;
+	delete _surface;
+}
+
+void Sprites::loadSprites(Common::ReadStream &spr_fl) {
+	byte magic[4];
+
+	while (true) {
+		if (spr_fl.read(magic, sizeof(magic)) == 0) {
+			break;
+		}
+		if (memcmp(magic, "SPRI", sizeof(magic))) {
```

Comment:
I am not sure how well it will work, but the common way would be:

```
  if (spr_fl.readUint32BE() != MKTAG('S','P','R','I'))
      error("Invalid sprite magic");
```